GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -81,11 +81,18 @@ public override void Stop()
             out UpdateFailureReason failureReason)
         {
             UpdateFailureCause failureCause = UpdateFailureCause.NoFailure;
+
+            // TODO(Mac): Add functional tests that include:
+            //     - Mode + content changes between commits
+            //     - Mode only changes (without any change to content, see issue #223)
+            ushort fileMode = this.FileSystemCallbacks.GitIndexProjection.GetFilePathMode(relativePath);
+
             Result result = this.virtualizationInstance.UpdatePlaceholderIfNeeded(
                 relativePath,
-                GetPlaceholderVersionId(),
-                ConvertShaToContentId(shaContentId),
+                ToVersionIdByteArray(GetPlaceholderVersionId()),

```

Comment:
This affects both Mac and Windows - can we just define a single readonly `byte[]` in the base class for the version id and just use that for all placeholders? This is essentially a constant, and right now we're paying the cost of allocating it for every single placeholder. And on Mac it's actually a double allocation.

Reply:
Good idea, I'll make the change

Reply:
I will define a readonly `byte[]` but it can't be shared as its length relies on the platform specific `VirtualizationInstance`.